lib/ansible/plugins/connection/ssh.py

```diff
@@ -542,7 +543,12 @@ def _build_command(self, binary, *other_args):
 
         key = self._play_context.private_key_file
         if key:
-            b_args = (b"-o", b'IdentityFile="' + to_bytes(os.path.expanduser(key), errors='surrogate_or_strict') + b'"')
+            b_identity_key = to_bytes(os.path.expanduser(key), errors='surrogate_or_strict')
+            identity_key_st = os.stat(b_identity_key)
+            if identity_key_st & (stat.S_IRWXG | stat.S_IRWXO):
+                raise AnsibleError("Private key file %s cannot be used as its file permissions are insecure."
+                                   " Permission should be 600." % key)
+            b_args = (b"-o", b'IdentityFile="' + b_identity_key + b'"')
```

Comment:
We have removed checks like this in the past.  One thing to note is that the key need not exist on the filesystem to work if it has already been loaded into the agent.  We would need to ensure that missing files are allowed.